gfx_scene.h

```diff
@@ -1886,39 +1716,170 @@ class GfxSceneInternal
         image_ref->height = (uint32_t)image_height;
         image_ref->channel_count = resolved_channel_count;
-        image_ref->bytes_per_channel = (uint32_t)1;
-        image_ref->format = gfxImageGetFormat(*image_ref);
-        uint8_t *data = image_ref->data.data();
-        uint8_t alpha_check = 255;  // check alpha
-        for(int32_t y = 0; y < image_height; ++y)
-            for(int32_t x = 0; x < image_width; ++x)
-                for(int32_t k = 0; k < (int32_t)resolved_channel_count; ++k)
-                {
-                    int32_t const dst_index = (int32_t)resolved_channel_count * (x + (image_height - y - 1) * image_width) + k;
-                    int32_t const src_index = channel_count * (x + y * image_width) + k;
-                    uint8_t const source = (k < channel_count ? image_data[src_index] : (uint8_t)255);
-                    if(k == 3) alpha_check &= source;
-                    data[dst_index] = source;
-                }
-        image_ref->flags = (alpha_check == 255 ? 0 : kGfxImageFlag_AlphaChannel);
+        image_ref->bytes_per_channel = bytes_per_channel;
+        image_ref->format = GetImageFormat(*image_ref);
+        if(bytes_per_channel == 1)
+        {
+            uint8_t *data = image_ref->data.data();
+            uint8_t alpha_check = 255;  // check alpha
+            for (int32_t y = 0; y < image_height; ++y)
+                for (int32_t x = 0; x < image_width; ++x)
+                    for (int32_t k = 0; k < (int32_t)resolved_channel_count; ++k) {
+                        int32_t const dst_index = (int32_t)resolved_channel_count * (x + y * image_width) + k;
+                        int32_t const src_index = channel_count * (x + y * image_width) + k;
+                        uint8_t const source = (k < channel_count ? image_data[src_index] : (uint8_t)255);
+                        if (k == 3) alpha_check &= source;
+                        data[dst_index] = source;
+                    }
+            image_ref->flags = (alpha_check == 255 ? 0 : kGfxImageFlag_AlphaChannel);
+        }
+        else
+        {
+            uint16_t *data = (uint16_t*)image_ref->data.data();
+            uint16_t alpha_check = 65535;   // check alpha
+            for (int32_t y = 0; y < image_height; ++y)
+                for (int32_t x = 0; x < image_width; ++x)
+                    for (int32_t k = 0; k < (int32_t)resolved_channel_count; ++k) {
+                        int32_t const dst_index = (int32_t)resolved_channel_count * (x + y * image_width) + k;
+                        int32_t const src_index = channel_count * (x + y * image_width) + k;
+                        uint16_t const source = (k < channel_count ? image_data[src_index] : (uint16_t)65535);
+                        if (k == 3) alpha_check &= source;
+                        data[dst_index] = source;
+                    }
+            image_ref->flags = (alpha_check == 65535 ? 0 : kGfxImageFlag_AlphaChannel);
+        }
```

Comment:
Loading image data now supports both 8b and 16b. The main difference though is that this loads images in the order that the data is contained in the source file, whereas previously it was loaded flipped. This new version is what is required for loading images as found in gltf files but might cause issues with obj files being flipped????

Reply:
I believe the issue is that obj's UV vertical origin is swapped compared to glTF's configuration.
So like for one format y=0 means top, for the other y=0 means bottom...
That's also why the image loading was "repeated" inside of the previous `glTF` loading code iirc.
One option could be adding a "flip_image" option to the image loading function? or try and invert the obj's UVs somehow?

Reply:
Just the vertical component of the UVs obviously.

Reply:
I think just having a `uv.y = 1.0f - uv.y` in the obj loader might suffice to fix the issue, bit annoying but it does seem like the less intrusive solution.

Reply:
OK, ill flip the uvs in the obj loader as I think thats probably the best solution (as it allows compressed textures to be used there as well)

Reply:
Ok great, let me know when this is done.

Reply:
Added the fix (seems pretty simple)